src/ast/irbuilderbpf.cpp

```diff
@@ -273,7 +273,7 @@ Value *IRBuilderBPF::CreateMapLookupElem(Value *ctx,
   CreateCondBr(condition, lookup_success_block, lookup_failure_block);
 
   SetInsertPoint(lookup_success_block);
-  if (type.IsAggregate())
+  if (type.IsAggregate() || type.IsTimestampTy())
```

Comment:
I think we should either put timestamp in IsAggregate or find a new name to describe them both (and all future async events which should be handled like this).

Reply:
I'd agree. I think `IsCompound` could be a nice name here since C++ seems to use similar term: [`std::is_compound`](https://en.cppreference.com/w/cpp/types/is_compound) and it includes array, tuple, class and so on (`Type::timestamp` seems like a class type to me).  Btw there could be some other options like `IsCollectionTy`, `IsComposite` and `IsCombined` though I'm not sure if there are as good.

Reply:
Discussed with @danobi and thought that it might be good to add a helper function instead.